src/imerg/imerg.py

```diff
@@ -0,0 +1,110 @@
+from datetime import datetime
+import logging
+import os
+import pandas as pd
+import requests
+from pathlib import Path
+from typing import Literal
+import xarray as xr
+from azure.storage.blob import StandardBlobTier
+
+from constants import IMERG_BASE_URL, SAS_TOKEN_DEV, CONTAINER_RASTER
+from src.utils.azure_utils import upload_file, upload_file_by_mode
+
+logger = logging.getLogger(__name__)
+logging.getLogger("azure.core.pipeline.policies.http_logging_policy").setLevel(
+    logging.WARNING
+)
+logging.getLogger("botocore.credentials").setLevel(logging.WARNING)
+
+
+def download(date: datetime = datetime.today(),
+             run: Literal["E", "L"] = "L",
+             version: int = 7,
+             save_raw: bool = False,
+             output_dir: Path = "",
+             mode="local"):
+    """
+        Downloads IMERG data for a given date range and saves it to a location in a configured container
```

Comment:
I think we have our docstrings formatted slightly differently, which we should probably standardize. I don't have strong opinions either way -- happy to modify mine if you like this better.

Reply:
Yes, I noticed 😅 Either way it's fine with me.

Reply:
No strong preference on my side either. In our old coding guidelines I believe it was specified to use `numpy`-style docstrings (i.e. the format here). So many of our past repos _should_ have this formatting.